### DISCUSSION TITLE
NPM command not found when deploy using NPM workspace repository

### DISCUSSION THREAD

**Discussion by mupinnn** [vercel/community#235] [Help] [answered]
Hi! I'm getting this `npm ERR! missing script: build` error when trying to deploy my Vite app from simple monorepo using NPM workspaces.

```
10:50:27.647 |> npm-workspaces@1.0.0 build:client /vercel/path1
10:50:27.647 | > npm run build --workspace=@packages/client
10:50:27.647 |  
10:50:27.928 | npm ERR! missing script: build
10:50:27.954 |  
10:50:27.954 | npm ERR! A complete log of this run can be found in:
10:50:27.955 | npm ERR!     /vercel/.npm/_logs/2022-01-21T03_50_27_928Z-debug.log
10:50:27.964 | npm ERR! code ELIFECYCLE
10:50:27.964 | npm ERR! errno 1
10:50:27.969 | npm ERR! npm-workspaces@1.0.0 build:client: `npm run build --workspace=@packages/client`
10:50:27.969 | npm ERR! Exit status 1
10:50:27.970 | npm ERR!
10:50:27.970 | npm ERR! Failed at the npm-workspaces@1.0.0 build:client script.
10:50:27.970 | npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
10:50:27.978 |  
10:50:27.979 | npm ERR! A complete log of this run can be found in:
10:50:27.979 | npm ERR!     /vercel/.npm/_logs/2022-01-21T03_50_27_970Z-debug.log
10:50:27.991 | Error: Command "npm run build:client" exited with 1
```

The directory structure:
![image](https://user-images.githubusercontent.com/46535277/150479724-90e77616-0b18-42a4-8c74-de46f603358f.png)

Root `package.json`:
```json
{
  "name": "npm-workspaces",
  "private": true,
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "build:client": "npm run build --workspace=@packages/client",
    "dev:server": "npm run start --workspace=@packages/server",
    "dev:client": "npm run dev --workspace=@packages/client",
    "dev": "concurrently \"npm run dev:server\" \"npm run dev:client\""
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "engines": {
    "npm": ">=7"
  },
  "workspaces": [
    "packages/server",
    "packages/client",
    "packages/shared"
  ],
  "devDependencies": {
    "concurrently": "^7.0.0"
  }
}
```

Client/Vite `package.json`:
```json
{
  "name": "@packages/client",
  "version": "0.0.0",
  "scripts": {
    "dev": "vite",
    "build": "vite build",
    "preview": "vite preview"
  },
  "dependencies": {
    "react": "^17.0.2",
    "react-dom": "^17.0.2",
    "socket.io-client": "^4.4.1"
  },
  "devDependencies": {
    "@vitejs/plugin-react": "^1.0.7",
    "vite": "^2.7.2"
  }
}
```

I'm running `npm run build:client` locally work smoothly as well as running `npm run build --workspace=@packages/client`. However, when I push it and deployment running, it always throws an error command not found; event though the command is present.

This is my current deployment setting:
![image](https://user-images.githubusercontent.com/46535277/150480183-a8a2ec18-4f0d-4a5b-a100-6863eac0ea67.png)
![image](https://user-images.githubusercontent.com/46535277/150480220-64160043-9fd3-4439-851e-fbb9d0ea0372.png)

**Note:**
- The reason why I set the output directory to `packages/client/dist` is that the Vite app lives there.
- For the root directory, I didn't set it to `packages/client` by reason of it can't get and build with the shared package that lives on `packages/shared`.

Is there any mistake with my deployment setting? let me know! Thank you

_(pardon my english T_T)_

**Comment by Pieparker** [comment, marked as answer]
Hey there! Are you still struggling with this? I suspect that it's [somewhat related to this](https://github.com/npm/cli/issues/1904#issuecomment-805458988) however I'm hardly a Workspaces expert, I'm afraid!

**Reply by mupinnn** [reply]
Hi! Thanks for replying to my issues right here. Actually, I got the answer back then early February but I didn't have time to try it out (just recently I can spend time tinkering this issue again).

I asked Vercel Support about the error and they say "npm workspaces are not currently supported on Vercel". I'm wondering because that time I've seen the official announcement from Vercel that they support npm v7 which also can use the workspace features of npm. Then I asked "how about yarn workspaces?" and they replied:

![Vercel support reply about npm workspace feature support and yarn workspace support on their platform](https://user-images.githubusercontent.com/46535277/158208005-25755ec1-3995-4c0d-8e5d-f73db9b25a1f.png)

So today I tried to use yarn workspace and it works like a charm!
